test/OperationTest.cpp

```diff
@@ -242,15 +242,24 @@ TEST(OperationTest, estimatesForCachedResults) {
 
     [[maybe_unused]] auto res = qet->getResult();
   }
-  // The result is now cached inside the static execution context, if we create
-  // the same operation again, the cost estimate is 0. The size estimate doesn't
+  // The result is now cached inside the static execution context. If we create
+  // the same operation again and `zero-cost-for-cached-subtrees` is set to
+  // `true`, the cost estimate should be zero. The size estimate does not
   // change (see the `getCostEstimate` function for details on why).
   {
+    RuntimeParameters().set<"zero-cost-for-cached-subtree">(true);
```

Comment:
This parameter should be reset to its original value. This makes a difference for the coverage build, where a single binary containing all the unit tests is built and so this flag will be set for all unit tests that follow until it is changed back. This can lead to unwanted side effects.

There's a function `setRuntimeParameterForTest` in `RuntimeParametersTestHelpers.h` that does just this.